core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMap.java

```diff
@@ -178,6 +191,26 @@ private void loadToUnsafe(DataFileFooter fileFooter, SegmentProperties segmentPr
         throw new RuntimeException(e);
       }
     }
+    // write the task level min/max row to unsafe memory store
+    if (null != taskMinMaxRow) {
+      addRowToUnsafeMemoryStore(taskMinMaxRow);
+    }
+  }
+
+  private void addRowToUnsafeMemoryStore(DataMapRow row) {
```

Comment:
I think better set some junk or default values exept for the minmax in case of last row. so that it will be easy to read and write